week7/MoonSeoIn/src/pages/Cart/Cart.tsx

```diff
@@ -1,48 +1,71 @@
+import { useNavigate } from "react-router-dom";
 import CartItem from "../../components/CartItem/CartItem";
 import CartSummary from "../../components/CartSummary/CartSummary";
+import useCartStore from "../../pages/Store/useCartStore";
 
 const Cart = () => {
-  const store = {
-    name: "샐로리 한남점",
-    minDeliveryPrice: 13000,
-    deliveryFee: 2000,
+  const navigate = useNavigate();
+  const { menus, store, clearCart, getTotalPrice } = useCartStore();
```

Comment:
오 저도 최근까지는 요렇게 가져다가 썼는데 이게 권장되지 않는 방식이라 하더라구요. 이렇게 한번에 가져오게 되면 예를 들어서 menus값이 변경되면 menus를 쓰지 않는 컴포넌트도 같이 리렌더링이 발생한다고 합니다. 
```suggestion
const menus = useCartStore((state) => state.menus);
const store = useCartStore((state) => state.store);
const clearCart = useCartStore((state) => state.clearCart);
// ...
```
그래서 위처럼 하나씩 가져오는게 좋습니다. 만약 이게 코드가 길어지고 안 예쁘다 싶으면 useShallow를 쓰는 방법도 있습니다. useShallow는 zustand에서 selector로 가져온 객체의 얕은 비교를 자동으로 적용해주는 아이인데 아래처럼 쓸 수 있습니다.
```suggestion
const { menus, store, clearCart, getTotalPrice } = useCartStore(
  useShallow((state) => ({
    menus: state.menus,
    store: state.store,
    // ...
  }))
);
```

https://github.com/pmndrs/zustand?tab=readme-ov-file#recipes
zustand 깃허브인데 요기 readme파일에 recipes 부분에 이런 얘기가 조잘조잘 있더라구요. 참고해보면 좋을 것 같습니다